blog/2020-04-09-debezium-apicurio-registry.adoc

```diff
@@ -0,0 +1,616 @@
+      "default": null
+    },
+    {
+      "name": "after",
+      "type": [
+        "null",
+        "Value"
+      ],
+      "default": null
+    },
+...
+  ],
+  "connect.name": "dbserver1.inventory.customers.Envelope"
+}
+----
+
+The connector registration request also differs from the standard one in a handful of lines:
+
+[source,json]
+----
+...
+"key.converter": "io.apicurio.registry.utils.converter.AvroConverter", <1>
+"key.converter.apicurio.registry.url": "http://apicurio:8080", <2>
+"key.converter.apicurio.registry.converter.serializer":
+    "io.apicurio.registry.utils.serde.AvroKafkaSerializer", <3>
+"key.converter.apicurio.registry.converter.deserializer":
+    "io.apicurio.registry.utils.serde.AvroKafkaDeserializer", <3>
+"key.converter.apicurio.registry.global-id":
+    "io.apicurio.registry.utils.serde.strategy.AutoRegisterIdStrategy", <4>
+
+"value.converter": "io.apicurio.registry.utils.converter.AvroConverter", <1>
+"value.converter.apicurio.registry.url": "http://apicurio:8080", <2>
+"value.converter.apicurio.registry.converter.serializer":
+    "io.apicurio.registry.utils.serde.AvroKafkaSerializer", <3>
+"value.converter.apicurio.registry.converter.deserializer":
+    "io.apicurio.registry.utils.serde.AvroKafkaDeserializer", <3>
+"value.converter.apicurio.registry.global-id":
+    "io.apicurio.registry.utils.serde.strategy.AutoRegisterIdStrategy", <4>
+...
+----
+<1> Apicurio Avro converter is used as both key and value converter
+<2> Option points to the actual Apicurio registry endpoint
+<3> Prescribe which serializer and deserializer should be used by the converter
+<4> This setting ensures that it is posible to automatically register the schema id which is typical setting in Debezium deployment
+
+To consume the messages on the sink side we can, for example, demonstrate an Elasticsearch sink connector. The sink configuration will be again extended only with converter configuration and the sink connector can consume from Avro-enabled topics without any other changes needed.
+
+[source,json]
+----
+{
+    "name": "elastic-sink",
+    "config": {
+        "connector.class": "io.confluent.connect.elasticsearch.ElasticsearchSinkConnector",
+        "tasks.max": "1",
+        "topics": "customers",
+        "connection.url": "http://elastic:9200",
+        "transforms": "unwrap,key",
+        "transforms.unwrap.type": "io.debezium.transforms.ExtractNewRecordState",
+        "transforms.unwrap.drop.tombstones": "false",
+        "transforms.key.type": "org.apache.kafka.connect.transforms.ExtractField$Key",
+        "transforms.key.field": "id",
+        "key.ignore": "false",
+        "type.name": "customer",
+        "behavior.on.null.values": "delete",
+        "key.converter": "io.apicurio.registry.utils.converter.AvroConverter",
+        "key.converter.apicurio.registry.url": "http://apicurio:8080",
+        "key.converter.apicurio.registry.converter.serializer": "io.apicurio.registry.utils.serde.AvroKafkaSerializer",
```

Comment:
Can you wrap and indent the very long lines as I did in other examples?

Reply:
Done